### PR TITLE
refactor: Auth options refactor

### PR DESCRIPTION
**Pull request by Dhruwang**
## What does this PR do?

Refactored authOptions to use our internal services

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

Create account

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [How we Code at Formbricks](<[https://github.com/formbricks/formbricks/blob/main/CONTRIBUTING.md](https://formbricks.com/docs/contributing/how-we-code)>)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues
- [x] First PR at Formbricks? [Please sign the CLA!](https://formbricks.com/clmyhzfrymr4ko00hycsg1tvx) Without it we wont be able to merge it 🙏

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Formbricks Docs if changes were necessary

### CODE REVIEW

File: packages/lib/authOptions.ts [hunk 1]
```diff
@@ -205,119 +211,104 @@ export const authOptions: NextAuthOptions = {
...
+          },
+          {
+            name: "email",
+            description: "The email of the person",
+            type: "automatic" as "automatic",
+          },
+        ];
+
+        const registerUserAndSetupTeam = async () => {
+          const userProfile = await createProfileWithProviders({
```
Comment (mattinannt) [inline packages/lib/authOptions.ts:277]: I think for most of the fields we can use and extend the input type of `createProfile` and we don't need a separate service for this.
For the account we should just create a new service `createAccount` that we can use to create a new account with the id of the newly created as userId

File: packages/lib/authOptions.ts [hunk 2]
```diff
@@ -205,119 +211,104 @@ export const authOptions: NextAuthOptions = {
...
-          },
-          include: {
-            memberships: true,
-          },
-        });
+            accounts: [{ ...account }],
+          });
+          const team = await createTeam({ name: userProfile.name + "'s Team" });
+          const product = await createProduct(team.id, { name: "My Product" });
+          const productionEnvironment = await createEnvironment(product.id, { type: "production" });
```
Comment (mattinannt) [inline packages/lib/authOptions.ts:288]: We don't need to create the environment and actionClasses as this is already done by the `createProduct` service that creates the development and production environment :-)

File: packages/lib/authOptions.ts [hunk 3]
```diff
@@ -205,119 +211,104 @@ export const authOptions: NextAuthOptions = {
...
+              environmentId: developmentEnvironment.id,
+            });
+          });
+          devAttributeClasses.forEach(async (attributeClass) => {
+            await createAttributeClass(productionEnvironment.id, attributeClass.name, attributeClass.type);
+          });
+          await createMembership(team.id, user.id, { role: "owner" });
+        };
+
+        await registerUserAndSetupTeam();
```
Comment (mattinannt) [inline packages/lib/authOptions.ts:311]: why did you put `registerUserAndSetupTeam()` into a function and didn't call it directly as it's only used in this place :-) 
I think if the function gets simpler without the dev and prod environment setup we can just put the function content here and have good readability

**Review by mattinannt — CHANGES_REQUESTED** [review]
@Dhruwang thanks for the PR :-)💪 I only have a few suggestions how to make this more generic with an account service plus we don't need to create the environments and action classes as this is already done by the product service 😊

**Comment by mattinannt** [comment]
@Dhruwang unfortunately I'm still getting this error when signing up in with Github:

```
@formbricks/web:go: Validation failed for {"_def":{"unknownKeys":"strip","catchall":{"_def":{"typeName":"ZodNever"}},"typeName":"ZodObject"},"_cached":{"shape":{"userId":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"type":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"provider":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"providerAccountId":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"access_token":{"_def":{"innerType":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"typeName":"ZodNullable"}},"refresh_token":{"_def":{"innerType":{"_def":{"innerType":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"typeName":"ZodNullable"}},"typeName":"ZodOptional"}},"expires_at":{"_def":{"innerType":{"_def":{"checks":[],"typeName":"ZodNumber","coerce":false}},"typeName":"ZodNullable"}},"scope":{"_def":{"innerType":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"typeName":"ZodNullable"}},"token_type":{"_def":{"innerType":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"typeName":"ZodNullable"}},"id_token":{"_def":{"innerType":{"_def":{"checks":[],"typeName":"ZodString","coerce":false}},"typeName":"ZodNullable"}}},"keys":["userId","type","provider","providerAccountId","access_token","refresh_token","expires_at","scope","token_type","id_token"]}}: [
@formbricks/web:go:   {
@formbricks/web:go:     "code": "invalid_type",
@formbricks/web:go:     "expected": "number",
@formbricks/web:go:     "received": "undefined",
@formbricks/web:go:     "path": [
@formbricks/web:go:       "expires_at"
@formbricks/web:go:     ],
@formbricks/web:go:     "message": "Required"
@formbricks/web:go:   },
@formbricks/web:go:   {
@formbricks/web:go:     "code": "invalid_type",
@formbricks/web:go:     "expected": "string",
@formbricks/web:go:     "received": "undefined",
@formbricks/web:go:     "path": [
@formbricks/web:go:       "id_token"
@formbricks/web:go:     ],
@formbricks/web:go:     "message": "Required"
@formbricks/web:go:   }
```

**Comment by Dhruwang** [comment]
Did google worked ?

**Comment by mattinannt** [comment]
> Did google worked ?

Didn't try; don't have google setup locally at the moment ;-)

**Pull request merged**